src/metrics/listener.ts

```diff
@@ -50,13 +52,28 @@ export interface MetricsConfig {
 export class MetricsListener {
   private currentProcessor?: Promise<Processor>;
   private apiKey: Promise<string>;
+  private statsDClient?: StatsD;
+  private isAgentRunning?: boolean = undefined;
 
   constructor(private kmsClient: KMSService, private config: MetricsConfig) {
     this.apiKey = this.getAPIKey(config);
   }
 
-  public onStartInvocation(_: any) {
+  public async onStartInvocation(_: any) {
+    if (this.isAgentRunning === undefined) {
+      this.isAgentRunning = await isAgentRunning();
```

Comment:
So we would make the request for non-extension users as well? Could we benefit from short-circuit this with an env var and avoid the call?

Reply:
That's a good point, Remy did suggest we add an env var to control this as well. I think we should revisit for all the layers.